### ISSUE TITLE
We need a way to lookup OWNERs files

### ISSUE THREAD

**Issue by castrojo** [kubernetes/community#4018]
This will take lots of discussion so I'll just start it. It's difficult for me to keep track of all the things I am either an approver or reviewer of, it'd be nice if we could have a script or tool that let us do some lookups. Something like this (just spitballing) 

```
$ ./owners.sh castrojo
Org member: Yes

Reviewer (2):
github.com/kubernetes/community/foo
github.com/kubernetes/community/bar

Approver (1):
github.com/kubernetes/community/baz

Emeritus: (1):
github.com/kubernetes/community/baz

Warning: User is approver for over X amount of OWNERs files
```

/sig contributor-experience

**Comment by tpepper** [comment]
This should be a straight forward tool for somebody who's looking for some initial contributions and it's quite valuable to us to have.  With OWNERS_ALIASES the spread of who's where is really hard to follow.  Prow can do something similar on a PR, so maybe there's even some reusable code.  But from a contribex standpoint I think a standalone tool's useful for us to measure the sustainability of our ownership hierarchy and look for single points of failure.

**Comment by parispittman** [comment]
I think @mrbobbytables @spiffxp and others are working on this

**Comment by justaugustus** [comment]
Maybe we've got some prior art to leverage, like Hound?
e.g., https://cs.k8s.io/?q=justaugustus&i=nope&files=OWNERS&repos=

I _think_ @dims worked on this.

**Comment by justaugustus** [comment]
I either mentioned or forgot to mention on the GH Admin subproject call earlier...
Something like this would be super useful for gleaning legitimate sponsors for new membership requests.

**Comment by castrojo** [comment]
Yes, cs.k8s.io is what I use now, but what I was thinking about other possibilities outside of code searching in the future where we can leverage other APIs, etc, like say the ability to just see `Signed CLA: Yes` or something.

**Comment by nikhita** [comment]
xref: #1808

**Comment by mrbobbytables** [comment]
Writing a tool to do this over one or more cloned repos wouldn't be too difficult -- A service would be a bit more work.

With regard to searching cs.k8s.io, a query for `OWNERS` can turn up a lot of unneeded results centered around vendor directories. This will likely solve itself in time, but for now filtering out what you can is useful.

This is a bit more complicated, but helps eliminate them -- (thanks @BenTheElder for helping refine this)
`(^[^v].*/OWNERS$)|(^.[^e].*/OWNERS$)|(^..[^n].*/OWNERS$)|(^...[^d].*/OWNERS$)|(^....[^o].*/OWNERS$)|(^.....[^r].*/OWNERS$)|(^......[^/].*/OWNERS$)|(^[^v][^e][^n][^d][^o][^r]/.*OWNERS$)|^OWNERS_ALIASES$|^OWNERS$`

It becomes a pretty nasty url 😬 :
https://cs.k8s.io/?q=&i=fosho&files=.*(%5E%5B%5Ev%5D.*OWNERS%24)%7C(%5Ev%5B%5Ee%5D.*OWNERS%24)%20%7C(%5Eve%5B%5En%5D.*OWNERS%24)%7C(%5Even%5B%5Ed%5D.*OWNERS%24)%7C(%5Evend%5B%5Eo%5D.*OWNERS%24)%7C(%5Evendo%5B%5Er%5D.*OWNERS%24)%7C(%5E%5B%5Ev%5D%5B%5Ee%5D%5B%5En%5D%5B%5Ed%5D%5B%5Eo%5D%5B%5Er%5D%2F.*OWNERS%24)%7C%5EOWNERS%24%7C%5EOWNERS_ALIASES%24&repos=


Maybe create a bit.ly link or k8.io short url specifically for it? like bit.ly/k8s-owners or something along those lines?

**Comment by nikhita** [comment]
Can this be closed now?

We have `https://go.k8s.io/owners/${GITHUB_USER}` now. Thanks @mrbobbytables!! :)

https://groups.google.com/forum/#!topic/kubernetes-dev/t96WvmdacUc
